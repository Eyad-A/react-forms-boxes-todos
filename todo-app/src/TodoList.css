.todolist {
    text-align: center;    
    width: 500px;	
	background: #f4f7f8;
	margin: 30px auto;
	padding: 20px;	
	border-radius: 8px;
	font-family: Georgia, "Times New Roman", Times, serif;
}

.todolist fieldset{
	border: none;
}

.todolist legend {
	font-size: 18px;
	margin-bottom: 10px;
}

.todolist label {
	display: block;
	margin-bottom: 8px;
}

.todolist input[type="text"] {
    font-family: Georgia, "Times New Roman", Times, serif;
	background: rgb(180, 180, 180);
	border: none;
	border-radius: 4px;
	font-size: 18px;
	margin: 0;
	outline: 0;
	padding: 10px;
	width: 100%;
	box-sizing: border-box; 
	-webkit-box-sizing: border-box;
	-moz-box-sizing: border-box; 	
	color:#ffffff;
	-webkit-box-shadow: 0 1px 0 rgba(0,0,0,0.03) inset;
	box-shadow: 0 1px 0 rgba(0,0,0,0.03) inset;
	margin-bottom: 30px;
}

.todolist input[type="text"]:focus {
    background: #0000003a;
}

.todolist ul {
    text-align: left !important;
    width: 400px;
    border-spacing: 10px 10px;	
	background: #f4f7f8;
	margin: 5px auto;
	padding: 0;
	background: #f4f7f8;
    border-radius: 8px;
    text-align: center;
	font-family: Georgia, "Times New Roman", Times, serif;
}

.todolist li {
    list-style-type: none;
    font-size: 18px;
}