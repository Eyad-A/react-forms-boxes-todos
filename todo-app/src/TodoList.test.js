import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, task = "Finish bootcamp") {
    const taskInput = todoList.getByLabelText("Task:");
    fireEvent.change(taskInput, { target: { value: task }});
    const submitButton = todoList.getByText("Add a new todo");
    fireEvent.click(submitButton);
}

it("renders without crashing", function() {
    render(<TodoList />);
});

it("should match the snapshot", function() {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo", function() {
    const list = render(<TodoList />);
    addTodo(list);
    expect(list.getByText("Finish bootcamp")).toBeInTheDocument();
});