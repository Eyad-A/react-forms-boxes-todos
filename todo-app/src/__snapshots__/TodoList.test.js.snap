// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
      </div>
      <button>
        Add a new todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
